src/Requests.php

```diff
@@ -467,11 +467,21 @@ public static function request($url, $headers = [], $data = [], $type = self::GE
 			$transport    = self::get_transport($capabilities);
 		}
 
-		$response = $transport->request($url, $headers, $data, $options);
+		try {
+			$response = $transport->request($url, $headers, $data, $options);
+
+			$options['hooks']->dispatch('requests.before_parse', [&$response, $url, $headers, $data, $type, $options]);
 
-		$options['hooks']->dispatch('requests.before_parse', [&$response, $url, $headers, $data, $type, $options]);
+			$parsed_response = self::parse_response($response, $url, $headers, $data, $options);
+		} catch (Exception|InvalidArgument $e) {
```

Comment:
Uh oh.. multi-catch is PHP 7.1+ and this library still has a minimum supported PHP version of 5.6.

Reply:
But isn't the `Exception` class being thrown the one from Requests itself ? which the `InvalidArgument` class extends from ?

Reply:
`Exception` here is `\WpOrg\Requests\Exception`, which extends `\Exception`
`InvalidArgument` is `\WpOrg\Requests\Exception\InvalidArgument`, which extends `InvalidArgumentException`, which extends `\LogicException`, which extends `\Exception`

Reply:
Ah, yes, of course. In that case, I think the original code using two `catch()` statements is probably best.

Reply:
Alright, that's what I went with :)